Workflows

ancouli opened the pull request:
<!-- This PR fixes #NUMBER_OF_THE_ISSUE, and fixes #NUMBER_OF_THE_ISSUE -->

## Description

<!--
Include a concise description of the changes (bug or feature), 
its impact, along with a summary of the solution
-->

Unifies workflow actions for pushes to all branches (including master) and for pull requests from forks and from repo:
- Unit tests are run on all os with jdk 17 and 21
- Integration tests are run on ubuntu with jdk 17 if the workflow was not triggered by a fork
- JaCoCo and Spotbugs are run on ubuntu with jdk 17
- SonarQube analysis is performed for master branch
- SonarCloud analysis is performed in all cases
- An additional workflow is needed for SonarCloud analysis for PRs from forks. It will run if the first workflow completes successfully and a PR was originated in a fork. In all other cases it will be skipped. It needs the data from the first workflow. The data will be passed to the second workflow via an artifact. The artifact is uploaded in the first workflow if the workflow was triggered by a PR from a fork. It is retained for 1 day.
- A SAPL Server LT Image will be pushed to the repo if the workflow is triggered by the master branch
- A test report is published if the workflow was not triggered by a PR from a fork
- For each workflow in the repository the workflow runs will be deleted if they are older than 1 day. A minimum of 6 runs for each workflow will always be kept.
- Deployment to maven central will be performed if the workflow was triggered from the master branch.
- Throughout the main workflow PRs from Dependabot are treated the same as PRs from a fork, with the exception that no SonarCloud analysis will be run. Although PRs from Dependabot are generated in the same repo, access to secrets is not possible.
- The main workflow can be triggered manually.
- Tests on all other os/jdk will continue even if one of the matrix builds fails.

Updates build_documentation.yml, dependency-review.yml and scorecard.yml.


## Checklist

<!--
Fill out and perform the following checklist before publishing the PR for review.
-->

* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.
* [ ] All new and existing tests passed.
* [ ] My code follows the code style of this project.
* [x] I ran all checks which produced no new errors or warnings for my changes.
* [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

<!-- 📛📛📛
All pull requests go through pipelines and careful peer review to ensure the quality of the code, but this does not exempt developers from delivering good code. 
Write code to the best of your ability while considering best practices, guidelines, and requirements.

If it fixes any current issues please let us know this way:
Uncomment the comment above the "description", then add your number of issues after the "#".
Example: # **This pull request fixes #NUMBER_OF_THE_ISSUE issue**
If there are multiple issues to be closed with the merge of this pull request
please do it like so: **This pull request fixes #NUMBER_OF_THE_ISSUE, fixes #NUMBER_OF_THE_ISSUE and fixes #NUMBER_OF_THE_ISSUE issue**.
For more information on closing issues using keywords, please check https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#closing-multiple-issues
📛📛📛 -->

heutelbeck commented:
As this is specifically about the checks, why is this pull request hanging on waiting for Unit Tests (JDK 17 on ubuntu-latest) ?

ancouli commented:
Maybe it's because GitHub takes the workflow files from the PR and I deleted the workflow file `pull_request_fork.yaml` in my PR and replaced it by `build.yml`?
